apps/docs-snippets/test/fixtures/abi/encode-and-decode.jsonc

```diff
@@ -1,9 +1,11 @@
 // #region encode-and-decode-2
 {
+  "abiVersion": "1",
+  "specVersion": "1",
```

Comment:
Based on the [new spec](https://github.com/FuelLabs/fuel-specs/pull/596) and [incoming PR](https://github.com/FuelLabs/sway/pull/6254):
 - the `abiVersion` property was removed (unnecessary)
 - the `encoding` was renamed to `encodingVersion` (can be done in this PR as well)

We should have a basic switch approach (or similar) in place to lock the pattern for dealing with [future] different versions of `specVersion`, such as we have for `encodingVersion`.

@danielbate Thoughts?

https://github.com/FuelLabs/fuels-ts/blob/2ded63ea7a61bcdf3fffee845024136f7f784bdf/packages/abi-coder/src/encoding/strategies/getCoderForEncoding.ts#L9-L26

Reply:
A switch/strategy SGTM. It's been sound through development.